internal/run/convert.go

```diff
@@ -0,0 +1,190 @@
+package run
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/helm/helm-2to3/pkg/v3"
+	"github.com/helm/helm-2to3/pkg/common"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	utils "github.com/maorfr/helm-plugin-utils/pkg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pelotech/drone-helm3/internal/env"
+)
+
+type Convert struct {
+	*config
+	release string
+
+	deleteV2Releases bool
+	dryRun bool
+	kubeContext string
+	kubeConfig string
+	label string
+	releaseVersionsMax int
+	tillerNS string
+
+	cmd cmd
+}
+
+func NewConvert(cfg env.Config, tillerNS string, kubeConfig string) *Convert {
+	return &Convert{
+		config: newConfig(cfg),
+		release: cfg.Release,
+		deleteV2Releases: cfg.DeleteV2Releases,
+		dryRun: cfg.DryRun,
+		kubeContext: cfg.Context,
+		kubeConfig: kubeConfig,
+		label: cfg.TillerLabel,
+		releaseVersionsMax: cfg.ReleaseVersionsMax,
+		tillerNS: tillerNS,
```

Comment:
Why store both the `cfg` and specific members in the Convert struct?

Reply:
kubeConfig is not known by config, but it's constructed from other parameters and a template like in [this call ](https://github.com/mongodb-forks/drone-helm3/blob/master/internal/helm/plan.go#L95).

As for tillerNS, I agree, it probably should come from cfg instead. The main reason I use it as a parameter is to use`cfg.Namespace` as tiller namespace. In that way that parameter is not really needed. But even in that case, we should handle it differently

Reply:
Ah, I see `newConfig` is an overloaded term that strips out information from the env.Config

Reply:
yes, it's defined in a different file, but is part of the run package